website/content/docs/ecs/manual/secure-configuration.mdx

```diff
@@ -23,60 +23,150 @@ the necessary components of the task definition for Consul on ECS.
+| `-name`         | string  | A name of your choice. Must be unique among all auth methods.                                                                     |
+| `-description`  | string  | A description of your choice.                                                                                                     |
+| `-config`       | string  | A JSON string containing the [configuration](/docs/security/acl/auth-methods/aws-iam#config-parameters) for the the auth method.  |
+
+In the `-config` option, the following fields are required:
+
+| Field                     | Type    | Description                                                                                               |
+| ------------------------- | ------- | --------------------------------------------------------------------------------------------------------- |
+| `BoundIAMPrincipalArns`   | list    | The list of trusted IAM roles. We recommend using a wildcard to trust IAM roles at a particular path.     |
+| `EnableIAMEntityDetails`  | boolean | Must be true so that the auth method can retrieve IAM role details, such as the role path and role tags.  |
+
+The following creates the Consul role and policy used for client tokens.
+
+```shell
+consul acl policy create -name consul-ecs-client-policy -rules 'node_prefix "" {
+ policy = "write"
 }
 service_prefix "" {
   policy = "read"
 }
+'
+
+consul acl role create -name consul-ecs-client-role -policy-name consul-ecs-client-policy
 ```
 
-This policy allows `node:write` for any node name, which is necessary because the Consul node
-names on ECS are not known until runtime.
+The following binding rule associates the `consul-ecs-client-role` with each token created by a
+successful login to this auth method instance.
 
-### Create service tokens
+```shell
+consul acl binding-rule create -method iam-ecs-client-token \
+   -description 'Bind a role for Consul clients on ECS' \
+   -bind-type role \
+   -bind-name consul-ecs-client-role
+```
 
-Service tokens should be associated with a [service identity](/docs/security/acl#service-identities).
-The service identity includes `service:write` permissions for the service and sidecar proxy.
+### Auth Method for Service Tokens
 
-The following example shows how to use the Consul CLI to create a service token for a service named `example-client-app`:
+The following uses the Consul CLI to create an instance of the auth method
+for service tokens. This configures the auth method to associate a service identity
+to each token created during login to this auth method instance.
 
 ```shell
-consul acl token create -service-identity=example-client-app ...
+consul acl auth-method create -type aws-iam \
+    -name iam-ecs-service-token \
+    -description="AWS IAM auth method for ECS service tokens" \
+    -namespace-rule-selector 'entity_tags["consul.hashicorp.com.namespace"] != ""' \
+    -namespace-rule-bind-namespace '${entity_tags.consul.hashicorp.com.namespace}' \
```

Comment:
this only applies to ent right?

Reply:
Right! I'll split this into the non-ent / ent configs.